Toolchain/BuildIt.sh

```diff
@@ -372,6 +372,15 @@ popd
 pushd "$DIR"
     if [ "${TRY_USE_LOCAL_TOOLCHAIN}" = "y" ] ; then
         echo "::endgroup::"
+
+        # FIXME: Qemu currently needs a local patch for AArch64 testing.
+        #        It is included in Qemu 8.1; remove this when upgrading!
+        if [ "$ARCH" = "aarch64" ]; then
+            echo "::group::Building custom patched Qemu"
+            "$DIR/BuildQemu.sh"
```

Comment:
Do we have to do this in BuildIt.sh, or can we do it in the CI script directly?

Reply:
eh. I guess a better question is if the BuildQemu.sh script smart enough to skip building things it doesn't have to when the "cached archive" is present.

Reply:
We could look into using `cache@v3` again and remove the cache handling logic from the build scripts.

Reply:
#19094

Reply:
Ah using that new set of cache steps, you could simply have a separate build step to build/cache qemu that only depends on BuildQemu.sh and our patches, and bypass doing it in BuildIt.sh altogether.